### PR TITLE
#18990 refactoring a simpler way to do the key value for the page re…

### PR DESCRIPTION
**Pull request by jdotcms**
This is a simpler way to do the key value field, only per depend and for the page render instead of doing a general approach (which is a bit problematic)

### CODE REVIEW

**Review by erickgonzalez — APPROVED** [review]

**Review by freddyucv — CHANGES_REQUESTED** [review]
any test?

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/transform/DotTransformerBuilder.java [hunk 1]
```diff
@@ -125,6 +125,17 @@ public DotTransformerBuilder keyValueToMapTransformer(){
         return this;
     }
 
+    /**
+     * Url Content Map to Map Transformer
+     * @return
+     */
+    public DotTransformerBuilder urlContentMapTransformer(){
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/transform/DotTransformerBuilder.java:132]: I not understand what it is this method doing, I think it need a better doc

**Review by freddyucv — CHANGES_REQUESTED** [review]

**Review by nollymar — APPROVED** [review]

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/transform/DotTransformerBuilder.java (hunk 1 above)
Comment (jdotcms) [reply dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/transform/DotTransformerBuilder.java:132]: I am just following the same convention that is already done by @fabrizzio-dotCMS

**Comment by jdotcms** [comment]
> any test?

This one is part of an initial request, just making it simpler; the test for this code has been already merge before

**Pull request merged**